File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/controlsystems/SquID.java
package org.firstinspires.ftc.teamcode.controlsystems;

public class SquID {
    double kp;
    double kd;
    double ki;
    double currentPos;
    double targetPos;
    double error;
    public double calculatePower(double targetPos, double currentPos){
        this.currentPos=currentPos;
        this.targetPos=targetPos;
        this.error=this.currentPos - this.targetPos;
        return kp*Math.sqrt(Math.abs(error))*Math.signum(error);
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/controlsystems/SquID.kt
import org.firstinspires.ftc.teamcode.drivetrain.Vector
import kotlin.math.absoluteValue
import kotlin.math.sign
import kotlin.math.sqrt

fun SquID(error: Vector, kSq: Double): Vector {
    return error.norm() * sqrt(error.length * kSq)
}

fun SquID(error: Double, kSq: Double): Double{
    return error.sign * sqrt(error.absoluteValue * kSq)
}